main.py

```diff
@@ -17,28 +27,183 @@ class YAMLParam(TypedDict):
 def parse_yaml(yaml_path: str) -> dict[str, Quantity[float]]:
     with open(yaml_path, "r") as file:
         data = cast(list[YAMLParam], yaml.safe_load(file))
-
-    result: dict[str, timestamp | Quantity[float]] = {}
-
+    
+    result: dict[str, datetime | Quantity[float]] = {}
     for param in data:
         if param['unit'] == 'datetime':
-            result[param['name']] = datetime.timestamp(param['value'])
+            # Handle if value is already a datetime object or a string
+            if isinstance(param['value'], datetime):
+                result[param['name']] = param['value']
+            else:
+                result[param['name']] = datetime.fromisoformat(str(param['value']))
         else:
             result[param['name']] = param['value'] * UNIT_REGISTRY(param['unit'])
-
+    
     return result
 
+def run_simulation(m: VehicleModel, log_params: list[str]) -> pd.DataFrame:
+    # Total number of timesteps
+    total_steps = int(
+        (m.params["raceday_len"] / m.params["timestep"]).to("dimensionless").magnitude
+    )
+    
+    # Get start time from params, default to 9:00 AM if not specified
+    if "start_ts" in m.params:
+        current_time = m.params["start_ts"]
+    else:
+        current_time = datetime.now().replace(hour=9, minute=0, second=0, microsecond=0)
```

Comment:
`current_time = m.params.get("start_ts", datetime(2026, 7, 1, 9, 0, 0, 0))`

or something like that